ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/properties/FontSpec.kt

```diff
@@ -58,9 +60,50 @@ internal sealed interface FontSpec {
 internal fun Map<FontAlias, FontsConfig>.determineFontSpecs(
     resourceProvider: ResourceProvider,
 ): Map<FontAlias, FontSpec> {
+    val resourceFontFamilies = values
+        .toSet()
+        .map { it.android }
+        .filterIsInstance<FontInfo.Name>()
+        .mapNotNull { fontInfo -> fontInfo.family?.let { family -> fontInfo to family } }
+        .groupBy({ it.second }, { it.first })
+        .mapValues { (_, fontInfos) ->
+            val resourceIdsSeen = mutableSetOf<Int>()
+            fontInfos.mapNotNull { fontInfo ->
+                resourceProvider.getResourceIdentifier(name = fontInfo.value, type = "font")
+                    .takeIf { it != 0 && it !in resourceIdsSeen }
+                    ?.also { resourceIdsSeen.add(it) }
+                    ?.let {
+                        ResourceFontSpec(
+                            id = it,
+                            weight = fontInfo.weight,
+                            style = fontInfo.style?.toComposeFontStyle(),
+                        )
+                    }
+            }
+        }
+        .filterValues { it.isNotEmpty() }
+        .mapValues { (_, resourceFonts) ->
+            if (resourceFonts.size == 1) {
```

Comment:
I took the decission here to only check if the resource is an xml font if all the fonts in the same family point to the same resource. If you have different resources for different fonts in the same font family, it's very unlikely for it to be an xml I think. This way we avoid querying for these extra resource files in those cases.